### PR TITLE
Rewrite SimActivityRegistryEnroller with variadic types

### PR DESCRIPTION
**Pull request by davidlange6**
Should remove the last use of boost/mpl in cmssw.

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

**Comment by civanch** [comment]
@davidlange6 , do we really need this complicate signals?

**Comment by davidlange6** [comment]
You mean the original structure? I don’t know. Its now less complex than previously in my judgement.

File: SimG4Core/Notification/go.1 [hunk 1]
```diff
@@ -0,0 +1,276 @@
+>> Local Products Rules ..... started
```
Comment (makortel) [inline SimG4Core/Notification/go.1:1]: Is this file necessary?

Comment (davidlange6) [reply SimG4Core/Notification/go.1:1]: whoops! will fix that

**Comment by davidlange6** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by silviodonato** [comment]
+1
